springql-core/src/time.rs

```diff
@@ -71,42 +81,75 @@ impl Add for Duration {
     }
 }
 
-pub struct DateTime(chrono::DateTime<FixedOffset>);
+pub struct DateTime(time::OffsetDateTime);
 
 impl DateTime {
     pub fn parse_from_rfc3339(s: &str) -> Result<Self, TimeError> {
-        Ok(Self(chrono::DateTime::parse_from_rfc3339(s)?))
+        Ok(Self(time::OffsetDateTime::parse(
+            s,
+            &time::format_description::well_known::Rfc3339,
+        )?))
     }
 
     pub fn naive_utc(&self) -> NaiveDateTime {
-        NaiveDateTime(self.0.naive_utc())
+        NaiveDateTime(to_primitive(self.0))
     }
 }
 
 #[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize, new)]
-pub struct NaiveDateTime(#[serde(with = "datetime_format")] chrono::NaiveDateTime);
+pub struct NaiveDateTime(#[serde(with = "datetime_format")] time::PrimitiveDateTime);
 
-pub const MIN_DATETIME: NaiveDateTime = NaiveDateTime(chrono::naive::MIN_DATETIME);
-const FORMAT: &str = "%Y-%m-%d %H:%M:%S%.9f";
+pub const MIN_DATETIME: NaiveDateTime = NaiveDateTime(time::PrimitiveDateTime::MIN);
+
+fn format_description(format: bool) -> &'static [time::format_description::FormatItem<'static>] {
+    // port from chrono format : const FORMAT: &str = "%Y-%m-%d %H:%M:%S%.9f";
+    if format {
+        // formatting pattern must have 9 digits for subsecond
+        format_description!("[year]-[month]-[day] [hour]:[minute]:[second].[subsecond digits:9]")
+    } else {
+        // parsing pattern unlimited digits for subsecond (we have test case for: 10 digits subsecond )
```

Comment:
Sorry, the test case must be wrong...

Could you fix the test case(s)?

Reply:
And then you may just define

```rust
const FORMAT_DESCRIPTION = format_description!("[year]-[month]-[day] [hour]:[minute]:[second].[subsecond digits:9]");
```

Reply:
> "processing the input generously and outputting it strictly".

Basically, I agree with the policy but I didn't mean to apply it to the timestamp format.

Even if we allow arbitrary length `subsecond`, the format still is strict in that:

- year, month, and day are separated by `-`, `/` is not allowed
- day and hour separator is ` `, not `T`
- subsecond is always required (`.` cannot be omitted)

IMO, accepting wider range of timestamp format is a good feature but should be done in another PR.

> Isn't this parsing process processing external input?

Yes.

We are currently specifying the length of subsecond here: https://springql.github.io/learn-springql/pipeline/source-reader#source-data-format

> timestamp format must currently have 9-digit fractional parts